src/CMakeLists.txt

```diff
@@ -2,65 +2,78 @@ cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 
 project(wenet_text_processing VERSION 0.1)
 
-include_directories(
-${CMAKE_SOURCE_DIR}
-)
-set(CMAKE_VERBOSE_MAKEFILE on)
+# This option makes it easy to use the wenet-text-processing runtime as a sub-project
+# in a CMake build, with another project as the top-level project (i.e., wenet).
+# When using wenet-text-processibng as a sub-project, we should set THRAX=OFF
+# since libthrax is not a dependence to libtext_processor and we assume libfst
+# has been built in top-level project.
+option(THRAX "whether to build thrax" OFF)
```

Comment:
so thrax is also required at the top level when buiding as a sub project?

Reply:
no, all we need is openfst, see  https://github.com/wenet-e2e/wenet-text-processing/pull/12/files#diff-148715d6ea0c0ea0a346af3f6bd610d010d490eca35ac6a9b408748f7ca9e3f4R74-R75